python/dgl/nn/pytorch/conv/edgegraphconv.py

```diff
@@ -0,0 +1,293 @@
+                                   'causing silent performance regression. '
+                                   'Adding self-loop on the input graph by '
+                                   'calling `g = dgl.add_self_loop(g)` will resolve '
+                                   'the issue. Setting ``allow_zero_in_degree`` '
+                                   'to be `True` when constructing this module will '
+                                   'suppress the check and let the code run.')
+            if edge_weight is None:
+                raise DGLError("Edge weight is required for this layer type (EdgeGraphConv)")
+            assert edge_weight.shape[0] == graph.number_of_edges()
+            aggregate_fn = fn.u_mul_e('h', '_edge_weight', 'm')
+
+            # (BarclayII) For RGCN on heterogeneous graphs we need to support GCN on bipartite.
+            feat_src, feat_dst = expand_as_pair(feat, graph)
+            if self._norm in ['left', 'both']:
+                degs = graph.out_degrees().float().clamp(min=1)
+                if self._norm == 'both':
+                    norm = th.pow(degs, -0.5)
+                else:
+                    norm = 1.0 / degs
+                shp = norm.shape + (1,) * (feat_src.dim() - 1)
+                norm = th.reshape(norm, shp)
+                feat_src = feat_src * norm
+            graph.srcdata['h'] = feat_src
+
+            if weight is not None:
+                if self.weight is not None:
+                    raise DGLError('External weight is provided while at the same time the'
+                                   ' module has defined its own weight parameter. Please'
+                                   ' create the module with flag weight=False.')
+            else:
+                weight = self.weight
+
+            single_channel_rsts = []
+            for i in range(edge_weight.shape[1]):
+                graph.edata['_edge_weight'] = edge_weight[:, i]
+                graph.update_all(aggregate_fn, fn.sum(msg='m', out='rst'))
+                single_channel_rst = graph.dstdata["rst"]
+                if weight is not None:
+                    single_channel_rst = th.matmul(single_channel_rst, weight)
+                if self.bias is not None:
+                    single_channel_rst = single_channel_rst + self.bias
+                single_channel_rsts.append(single_channel_rst)
+                    
+            rst = th.cat(single_channel_rsts, axis=1)
+
+            if self._norm in ['right', 'both']:
+                degs = graph.in_degrees().float().clamp(min=1)
+                if self._norm == 'both':
+                    norm = th.pow(degs, -0.5)
+                else:
+                    norm = 1.0 / degs
+                shp = norm.shape + (1,) * (feat_dst.dim() - 1)
+                norm = th.reshape(norm, shp)
+                rst = rst * norm
+
+            if self._activation is not None:
+                rst = self._activation(rst)
+
+            return rst
```

Comment:
do we want to return the edge features here as well so they can be fed into the next layer?